File: stock_table/stock_table/report/monthly_sales/monthly_sales.py
# Copyright (c) 2013, frappe and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import msgprint, _

def execute(filters=None):

	conditions, filters = get_conditions(filters)
	columns = get_column()
	data = get_data(conditions,filters)
	return columns,data

def get_column():
	return [_("Date") + ":Date:120",
			_("Invoice No") + ":Link/Sales Invoice:120",
			_("Customer Names") + ":Link/Customer:150",
			_("Grand Total") + ":Currency:100",
			_("Outstanding Amount") + ":Currency:100",
			_("Status") + ":Data:100"]

def get_data(conditions,filters):
	invoice = frappe.db.sql("""select posting_date,name, customer_name, grand_total, outstanding_amount, status from `tabSales Invoice` where docstatus is not null %s order by posting_date asc;"""%conditions, filters, as_list=1)
	return invoice

def get_conditions(filters):
	conditions = ""
	if filters.get("from_date"): conditions += " and posting_date >= %(from_date)s"
	if filters.get("to_date"): conditions += " and posting_date <= %(to_date)s"
	if filters.get("type"): conditions = "and type = %(type)s"

	return conditions, filters

File: stock_table/stock_table/report/monthly_sales/monthly_sales.js
// Copyright (c) 2016, Hardik Gadesha and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Monthly Sales"] = {
	"filters": [
		{
			"fieldname":"from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.month_start(),
			"reqd": 1,
			"width": "60px"
		},
		{
			"fieldname":"to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.month_end(),
			"reqd": 1,
			"width": "60px"
		},
		{
			"fieldname":"type",
			"label": __("Invoice Type"),
			"fieldtype": "Select",
			"options": "Cash Invoice\nCredit Invoice",
			"width": "60px"
		}
	]
}
